src/utils/ReviverUtils.ts

```diff
@@ -0,0 +1,31 @@
+import { toBN } from "./BigNumberUtils";
+import { isDefined } from "./TypeGuards";
+import { Reviver } from "./TypeUtils";
+
+/**
+ * A reviver composer that takes a list of revivers and returns a single reviver that applies all of them.
+ * @param revivers The revivers to compose.
+ * @returns A single reviver that applies all of the revivers passed in. If no revivers are passed in, the identity
+ * function is returned.
+ */
+export function composeRevivers(...revivers: Reviver[]): Reviver {
+  // Filter out undefined revivers.
+  revivers = revivers.filter(isDefined);
+  // Return a single reviver that applies all of the revivers passed in. If no revivers are passed in, the identity
+  // function is implicitly returned.
+  return (key: string, value: unknown) => revivers.reduce((acc, reviver) => reviver(key, acc), value);
+}
+
+/**
+ * Reviver function that converts a stringified BigNumber object to a BigNumber object.
+ * @param _ The key to the value being revived. Unused.
+ * @param value The value being revived.
+ * @returns The revived value. If the value is not a stringified BigNumber object, it is returned as-is.
+ */
+export function objectWithBigNumberReviver(_: unknown, value: unknown): unknown {
+  if (value === null || typeof value !== "object") {
+    return value;
+  }
+  const resolvedValue = value as { type: string; hex: string };
+  return resolvedValue.type === "BigNumber" ? toBN(resolvedValue.hex) : resolvedValue;
```

Comment:
Would `BigNumber.isBigNumber()` be worthwhile here? https://docs.ethers.org/v5/single-page/#/v5/api/utils/bignumber/-%23-BigNumber--BigNumber--methods--inspection

Reply:
Unfortunately not. This can be exemplified in this function - the stringify/parsing isn't equivalent:
```
const bn = ethers.BigNumber.from(4);
const stringified = JSON.stringify(bn);
const parsed = JSON.parse(stringified);
console.log(ethers.BigNumber.isBigNumber(parsed)) <- prints false
```

Reply:
Thanks for the clear example!